src/test/java/org/hydev/mcpm/client/installer/InstallResultTypeTest_Failure.java

```diff
@@ -0,0 +1,100 @@
+package org.hydev.mcpm.client.installer;
+
+import org.hydev.mcpm.client.installer.input.InstallInput;
+import org.hydev.mcpm.client.installer.output.InstallResult;
+import org.hydev.mcpm.client.search.SearchPackagesType;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests the Install Interactor
+ */
+public class InstallResultTypeTest_Failure {
+
+    /**
+     * Test Setup: Create an empty directory before testing
+     */
+    @BeforeAll
+    public static void dirSetUp() {
+        File file = new File("plugins");
+        boolean isDirectoryCreated = file.mkdir();
+        System.out.println(isDirectoryCreated);
+        if (!isDirectoryCreated) {
+            deleteDir(file);  // Invoke recursive method
+            file.mkdir();
+        }
+    }
+
+    /**
+     * Empty file in the directory
+     */
+    public static void deleteDir (File dir){
+        File[] files = dir.listFiles();
+
+        for (File pluginFile : files) {
+            if (pluginFile.isDirectory()) {
+                deleteDir(pluginFile);
+            }
+            pluginFile.delete();
+        }
+    }
+
+    /**
+     * Test case: invalid input, this happends when input name is empty -> searchPackageresult is null
+     */
+    @Test
+    public void testInvalidInput() {
+        InstallSetUp installSetUp = InstallSetUp.getInstaller();
+        InstallInput installInput = new InstallInput("",
+                                                    SearchPackagesType.BY_NAME,
+                                                    false,
+                                                    true);
+        InstallInteractor installInteractor = installSetUp.getInstallInteractor();
+        List<InstallResult> listInstallResult = installInteractor.installPlugin(installInput);
+        InstallResult installResult = listInstallResult.get(0);
+        assertEquals(installResult.name(), "");
+        assert (installResult.type() == InstallResult.Type.SEARCH_INVALID_INPUT);
+    }
```

Comment:
You can probably forget about side effects here. Your install interactor doesn't write to anything, right?

Reply:
No, the install interactor doesn't write to anything >< I was thinking of just testing all of the possible cases that could happen hehe and ensuring why there's such an install result type SEARCH_INVALID_INPUT.